lib/fs/filesystem.go

```diff
@@ -42,6 +42,7 @@ type Filesystem interface {
 	Watch(path string, ignore Matcher, ctx context.Context, ignorePerms bool) (<-chan Event, <-chan error, error)
 	Hide(name string) error
 	Unhide(name string) error
+	IsHidden(name string) (bool, error)
```

Comment:
I think this should not be a property of the filesystem but a property of the FileInfo.

Reply:
Same like a file has a name, size, permissions, it has a hidden property.

Reply:
Yep. Add it to `FileIntf` and then the types that fulfill that (protocol fileinfo and db truncated file info).

Reply:
From what I see `BasicFileInfo` doesn't provide any direct access to methods which could allow to query for attributes. In that case, `IsHidden` would need to take an argument with some filesystem. Other solution would be to extend `BasicFileInfo` with a field 'hidden' and everywhere it is constructed we would need to query for this attribute; from what I found it happens only when returning from `(BasicFileSystem).Lstat()`, `(basicFile).Lstat` and `(BasicFileSystem).Stat()`.

Is any of those options what you meant?

Reply:
Oh, it seems you meant a different `FileInfo`... :)

Reply:
Seems like it would fit among the other stat results like size and permissions etc, yes.